src/qpx_mmc.cpp

```diff
@@ -0,0 +1,4244 @@
+      printf("\nTrack #  : %d\n", drive->media.track[i].n);
+      printf("Session #: %d\n", drive->media.track[i].session);
+      printf("Track mode    : %d\n", drive->media.track[i].track_mode);
+      printf("Data mode     : %d\n", drive->media.track[i].data_mode);
+      printf("Track start   : %d\n", drive->media.track[i].start);
+      printf("Next writable : %d\n", drive->media.track[i].next_writable);
+      printf("Free          : %d\n", drive->media.track[i].free);
+      printf("Size          : %d\n", drive->media.track[i].size);
+      printf("Last recorded : %d\n", drive->media.track[i].last);
+    }
+  return 0;
+}
+
+int read_capacity(drive_info *drive) {
+  unsigned char data[8];
+  memset(data, 0, 8);
+  drive->cmd[0] = SBC_READ_CAPACITY;
+  drive->cmd[9] = 0;
+  if ((drive->cmd.transport(READ, data, 8))) {
+    if (!drive->silent)
+      sperror("READ_CAPACITY", drive->err);
+    drive->media.capacity = 0;
+    drive->media.sectsize = 2048;
+  } else {
+    drive->media.capacity = ntoh32(data);
+    if (drive->media.capacity)
+      drive->media.capacity++;
+    drive->media.sectsize = ntoh32(data + 4);
+  }
+  lba2msf(drive->media.capacity, &drive->media.capacity_msf);
+  return 0;
+}
+
+int read_capacity_free(drive_info *drive) {
+  trk track;
+  if (drive->media.dstatus == 2) {
+    drive->media.capacity_free = 0;
+    return 0;
+  }
+
+  read_disc_information(drive);
+  if (!read_track_info(drive, &track, drive->media.tracks)) {
+    // printf("track %d free: %d\n",drive->media.tracks,track.free);
+
+    drive->media.capacity_free = track.free;
+    lba2msf(drive->media.capacity_free, &drive->media.capacity_free_msf);
+    return 0;
+  }
+  /*
+      if (drive->media.type & DISC_CD) {
+      //if (0) {
+              if (((drive->media.last_lead_out >> 24) & 0xFF ) == 0xFF) {
+                      drive->media.capacity_free = 0;
+                      lba2msf(drive->media.capacity_free,&drive->media.capacity_free_msf);
+                      return 0;
+              } else {
+                      drive->media.capacity_free = drive->media.last_lead_out -
+  drive->media.capacity - 150 - 2;
+                      lba2msf(drive->media.capacity_free,&drive->media.capacity_free_msf);
+                      return 0;
+              }
+      } else if (drive->media.type & DISC_DVD) {
+  //		track.n = drive->media.sessions+1;
+              read_track_info(drive, &track, drive->media.sessions+1);
+              drive->media.capacity_free = track.free;
+              lba2msf(drive->media.capacity_free,&drive->media.capacity_free_msf);
+              return 0;
+      }
+  */
+  drive->media.capacity_free = 0;
+  return 1;
+}
+
+int read_capacity_total(drive_info *drive) {
+  unsigned int len;
+  //	char		header[40];
+  unsigned int phsta = 0, phend = 0;
+  //	union { unsigned char _e[4+40],_11[4+256]; } dvd;
+  drive->media.capacity_total = 0;
+  int doffs;
+
+  if (drive->media.type & DISC_CD) {
+    msf lout;
+    int ilout;
```

Comment:
Codacy found an issue: [The scope of the variable 'ilout' can be reduced.](https://app.codacy.com/manual/jeremiahwilson777/qpxtool/pullRequest?prid=6191247)